.github/workflows/main.yml

```
@@ -50,3 +51,84 @@ jobs:
     with:
       branch_name: ${{ github.ref }}
 
+  e2e-tests:
+    runs-on: ubuntu-latest
+    env:
+      ACTIONS_RUNNER_DEBUG: true
+      ACTIONS_STEP_DEBUG: true
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Update and install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y sshpass python3-venv pipx make git
+          pipx install --include-deps ansible
+          pipx ensurepath
+          source ~/.bashrc
+
+      - name: Install Ansible Galaxy collections
+        run: |
+          ansible-galaxy collection install kubernetes.core ansible.utils community.docker
+
+      - name: Generate SSH Key
+        run: |
+          ssh-keygen -t rsa -b 2048 -f ~/.ssh/id_rsa -N "" -q
+          cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+
+      - name: Clone aether-onramp repository
+        run: |
+          git clone --recursive https://github.com/opennetworkinglab/aether-onramp.git
+
+      - name: Update files
+        run: |
+          IP_ADDR=$(ip a show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
+          echo "Extracted IP: ${IP_ADDR}"
+          echo "[all]" > aether-onramp/hosts.ini
+          echo "node1 ansible_host=${IP_ADDR} ansible_user=runner ansible_ssh_private_key_file=~/.ssh/id_rsa" >> aether-onramp/hosts.ini
+          echo "" >> aether-onramp/hosts.ini
+          echo "[master_nodes]" >> aether-onramp/hosts.ini
+          echo "node1" >> aether-onramp/hosts.ini
+          echo "" >> aether-onramp/hosts.ini
+          echo "[worker_nodes]" >> aether-onramp/hosts.ini
+          echo "" >> aether-onramp/hosts.ini
+          echo "[gnbsim_nodes]" >> aether-onramp/hosts.ini
+          echo "node1" >> aether-onramp/hosts.ini
+          sed -i "s/ens18/ens3/" aether-onramp/vars/main.yml
```

Based on line 86, this should be as follows
```suggestion
          sed -i "s/ens18/eth0/" aether-onramp/vars/main.yml
```